Sources/SwiftyTailwind/Downloader.swift

```diff
@@ -63,6 +73,24 @@ class Downloader: Downloading {
         let binaryPath = directory.appending(components: [expectedVersion, binaryName])
         if localFileSystem.exists(binaryPath) { return binaryPath }
         try await downloadBinary(name: binaryName, version: expectedVersion, to: binaryPath)
+        let checksumPath = directory.appending(components: [expectedVersion, Self.sha256FileName])
+        try await downloadChecksumFile(version: expectedVersion, to: checksumPath)
+        do {
+            let binaryChecksum = try Self.checksumValidator.generateChecksumFrom(binaryPath)
+            guard try Self.checksumValidator.compareChecksum(from: checksumPath, to: binaryChecksum) else {
+                
+                if numRetries < 5 {
+                    // retry download
+                    logger.error("Checksum validation failed. Attempt #\(numRetries + 1) to retry download...")
+                    numRetries += 1
+                    return try await download(version: version, directory: binaryPath)
+                } else {
+                    throw DownloaderError.checksumIsIncorrect
+                }
+            }
+        } catch {
```

Comment:
This will catch the `DownloaderError.checksumIsIncorrect` error. I recommend letting that one blow all the way up to the root.

Reply:
Also good shout, I've updated to rethrow the error if it's an incorrect checksum, let me know if any further revisions are needed!